crates/runtime/tests/oracle/mod.rs

```diff
@@ -0,0 +1,296 @@
+                "VAL_BINARY_FLOAT", "VAL_BINARY_DOUBLE", "VAL_RAW", "VAL_BLOB",
+                "VAL_BOOL"
+            ) VALUES (
+                3, NULL, NULL, NULL, NULL, NULL,
+                NULL, NULL, NULL, NULL,
+                NULL, NULL, NULL, NULL, NULL, NULL,
+                NULL, NULL, NULL, EMPTY_BLOB(),
+                NULL
+            )
+            "#,
+            &[],
+        )?;
+
+    // Commit
+    client.execute("COMMIT", &[])?;
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn oracle_integration_test() -> Result<(), anyhow::Error> {
+    let _tracing = init_tracing(Some("integration=debug,info"));
+
+    test_request_context()
+        .scope(async {
+            let running_container =
+                start_oracle_docker_container(ORACLE_DOCKER_CONTAINER, ORACLE_PORT)
+                    .await
+                    .map_err(|e| {
+                        tracing::error!("start_oracle_docker_container: {e}");
+                        e
+                    })?;
+            tracing::debug!("Container started");
+
+            let retry_strategy = FibonacciBackoffBuilder::new().max_retries(Some(5)).build();
+            retry(retry_strategy, || async {
+                init_oracle_db(ORACLE_PORT)
+                    .await
+                    .map_err(RetryError::transient)
+            })
+            .await
+            .map_err(|e| {
+                tracing::error!("Failed to initialize Oracle database: {e}");
+                e
+            })?;
+
+            let federated_ds = make_oracle_dataset("\"TEST_TABLE\"", "test_tbl", ORACLE_PORT);
+            let mut accelerated_ds = make_oracle_dataset("\"TEST_TABLE\"", "test_tbl_accelerated", ORACLE_PORT);
+            accelerated_ds.acceleration = Some(spicepod::acceleration::Acceleration::default());
+
+            let app = AppBuilder::new("oracle_integration_test")
+                .with_dataset(federated_ds)
+                // Verify that accelerated table can be successfully loaded
+                .with_dataset(accelerated_ds)
+                .build();
+
+            let rt = Runtime::builder()
+                .with_app(app)
+                .with_datafusion_configuration_fn(configure_test_datafusion)
+                .build()
+                .await;
+
+            let cloned_rt = Arc::new(rt.clone());
+
+            // Set a timeout for the test
+            tokio::select! {
+                () = tokio::time::sleep(std::time::Duration::from_secs(60)) => {
+                    return Err(anyhow::Error::msg("Timed out waiting for datasets to load"));
+                }
+                () = cloned_rt.load_components() => {}
+            }
+
+            run_and_snapshot_query(
+                &rt,
+                "describe test_tbl",
+                "schema",
+            )
+            .await?;
+
+            run_and_snapshot_query(
+                &rt,
+                r#"select 
+                    round("ID") as ID, 
```

Comment:
We round ID as it has very high scale and fractional part might fluctate, for example I see it returned as `1.0000000000006389`